src/activity_item_handling.cpp

```diff
@@ -1200,64 +1244,113 @@ static activity_reason_info can_do_activity_there( const activity_id &act, playe
         if( !veh ) {
             return activity_reason_info::fail( NO_ZONE );
         } else {
-            // find out if there is a vehicle part here we can remove.
-            std::vector<vehicle_part *> parts = veh->get_parts_at( src_loc, "", part_status_flag::any );
-            for( vehicle_part *part_elem : parts ) {
-                const vpart_info &vpinfo = part_elem->info();
-                int vpindex = veh->index_of_part( part_elem, true );
-                // if part is not on this vehicle, or if its attached to another part that needs to be removed first.
-                if( vpindex == -1 || !veh->can_unmount( vpindex ) ) {
-                    continue;
-                }
-                for( const npc &guy : g->all_npcs() ) {
-                    if( guy.disp_name() != p.disp_name() && guy.activity_vehicle_part_index != -1 &&
-                        guy.activity_vehicle_part_index == vpindex ) {
+            if( act == activity_id( "ACT_VEHICLE_DECONSTRUCTION" ) ) {
+                // find out if there is a vehicle part here we can remove.
+                std::vector<vehicle_part *> parts = veh->get_parts_at( src_loc, "", part_status_flag::any );
+                for( vehicle_part *part_elem : parts ) {
+                    const vpart_info &vpinfo = part_elem->info();
+                    int vpindex = veh->index_of_part( part_elem, true );
+                    // if part is not on this vehicle, or if its attached to another part that needs to be removed first.
+                    if( vpindex == -1 || !veh->can_unmount( vpindex ) ) {
                         continue;
                     }
-                }
-                // if the vehicle is moving or player is controlling it.
-                if( abs( veh->velocity ) > 100 || veh->engine_on ) {
-                    continue;
-                }
-                // dont have skill to remove it
-                std::map<skill_id, int> removal_skills = vpinfo.removal_skills;
-                for( const auto &e : removal_skills ) {
-                    bool hasSkill = p.get_skill_level( e.first ) >= e.second;
-                    if( !hasSkill ) {
+                    for( const npc &guy : g->all_npcs() ) {
+                        if( guy.disp_name() != p.disp_name() && guy.activity_vehicle_part_index != -1 &&
```

Comment:
Why compare the name here? If you want to know whether `guy` and `p` refer to the same NPC, compare their pointers:
```suggestion
                        if( &guy != &p && guy.activity_vehicle_part_index != -1 &&
```

Reply:
guy is an npc and p is a player, in this context. Worth a cast to player for guy, to compare?